### PR TITLE
chore(ci): retry flaky tasks

### PR DESCRIPTION
**Pull request by millotp**
## 🧭 What and Why

Some jobs on the CI fails a lot, for different reasons, we can add a bit of retry for the 2 macOS jobs, and for the release because maven is often down, or kotlin has a weird failure.

### CODE REVIEW

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -430,13 +430,14 @@ jobs:
           type: minimal
           language: kotlin
 
-      - name: remove previous build from linux
```
Comment (millotp) [inline .github/workflows/check.yml:433]: this didn't help

File: .github/workflows/check.yml [hunk 2]
```diff
@@ -622,7 +630,7 @@ jobs:
 
   push_and_release:
     runs-on: ubuntu-22.04
-    timeout-minutes: 15
+    timeout-minutes: 30
```
Comment (shortcuts) [inline .github/workflows/check.yml:633]: omgwtf

Comment (millotp) [reply .github/workflows/check.yml:633]: kotlin is taking 12 min to deploy and often fails, so weed need 30 min when it retries

**Review by shortcuts — APPROVED** [review]
gg nothing to add

**Pull request merged**